### DISCUSSION TITLE
How to set dark mode?

### DISCUSSION THREAD

**Discussion by PILIHU2022** [hyprwm/Hyprland#5867] [Q&A] [answered]
I use Arch LInux, and I want to set a global dark mode. Then the applications can all follow my system appearance. How can I do it?

**Comment by Aqa-Ib** [comment, marked as answer]
Write in hyprland.conf the following.

```
exec = gsettings set org.gnome.desktop.interface gtk-theme "YOUR_DARK_GTK3_THEME"   # for GTK3 apps
exec = gsettings set org.gnome.desktop.interface color-scheme "prefer-dark"   # for GTK4 apps

env = QT_QPA_PLATFORMTHEME,qt6ct   # for Qt apps
```


Then reboot hyprland, open qt6ct and choose a dark theme for Qt apps.

**Reply by abdeoliveira** [reply]
This worked for me for some apps (lyx, for example). Others are still in the bright side (e.g. gnome-calculator and timeshift).

**Reply by hollisticated-horse** [reply]
for gtk4 apps this does not change anything

**Reply by abdeoliveira** [reply]
Yes, that's what I said.

**Reply by Aqa-Ib** [reply]
That is probably because your portals are not working properly.

Install these packages:
+ xdg-desktop-portal-gtk
+ xdg-desktop-portal-hyprland

Then create the file `~/.config/xdg-desktop-portal/hyprland-portals.conf` with the following contents:
```
[preferred]
default=hyprland;gtk
```

Fianlly, reboot the computer and try.

**Reply by hollisticated-horse** [reply]
Ha nice, your method worked. Quick question, does that mean that it's a good idea to also have xdg-desktop-portal-kde on the machine ? Or is it a GTK thing.

**Reply by Aqa-Ib** [reply]
With the hyprland and gtk portals you are good to go. You don't need more than those two at the moment. 

The kde portal is made for plasma desktop specifically, it depends on plasma-desktop in fact. There is also a gnome portal made for gnome desktop. The gtk portal is a generic one that every environment can use with few dependencies.

**Reply by rtgiskard** [reply]
`gsettings` seems to write the settings to file, is it necessary to set every time on hyprland launch?

**Reply by Aqa-Ib** [reply]
No, it isn't necessary. I use it for hot realoding the theme in session.

**Reply by hieuft** [reply]
> That is probably because your portals are not working properly.
> 
> Install these packages:
> 
> * xdg-desktop-portal-gtk
> * xdg-desktop-portal-hyprland
> 
> Then create the file `~/.config/xdg-desktop-portal/hyprland-portals.conf` with the following contents:
> 
> ```
> [preferred]
> default=hyprland;gtk
> ```
> 
> Fianlly, reboot the computer and try.

I did the same thing and added 2 `gsetting` lines above, but `nautilus` still uses the light theme

**Reply by Aqa-Ib** [reply]
Nautilus uses GTK4. Try executing the command `gsettings set org.gnome.desktop.interface color-scheme "prefer-dark"` in your terminal emulator. If it does not work, probably your gtk portal is not working.

**Reply by Klusio19** [reply]
> That is probably because your portals are not working properly.
> 
> Install these packages:
> 
> * xdg-desktop-portal-gtk
> * xdg-desktop-portal-hyprland
> 
> Then create the file `~/.config/xdg-desktop-portal/hyprland-portals.conf` with the following contents:
> 
> ```
> [preferred]
> default=hyprland;gtk
> ```
> 
> Fianlly, reboot the computer and try.

For GTK3 apps, `gsettings set org.gnome.desktop.interface gtk-theme "YOUR_DARK_GTK3_THEME"` works fine.
But 
it still does NOT work for GTK4 apps. However similar content of the file works for GTK4 apps for me as follows:
```
[preferred]
default=gtk;wlr
```

**Reply by StevenIReedIII** [reply]
I changed it from `exec` to `exec-once` and it worked fine

**Reply by tony-sol** [reply]
not worked for me, apps (like chrome and wezterm) are still in light mode :c

**Comment by ildar** [comment]
Awesome! thanks a lot!

**Reply by Aqa-Ib** [reply]
You are welcome :)

**Comment by rtgiskard** [comment]
I'm really puzzled on the case:

conf for hyprland:
```

env = HYPRCURSOR_THEME, Bibata-Modern-Classic
env = HYPRCURSOR_SIZE, 22
env = XCURSOR_THEME, Bibata-Modern-Classic
env = XCURSOR_SIZE, 22
env = GTK_THEME, Adwaita-dark

exec-once = waybar
exec-once = fcitx5 -d
exec-once = systemctl restart --user foot-server.socket

bind = $mod, N, exec, xdg-open ~            # nautilus
bind = $mod, R, exec, $(pidof -q wofi) && pkill wofi || wofi
```

Now I can start `nautilus` in several way:

```
1. hypr exec -> waybar -> wofi -> nautilus: -> light
2. hypr keybind -> wofi -> nautilus: -> light
3. hypr keybind -> xdg-open ~: -> dark
4. hypr keybind -> wofi -> foot [hypr env set] -> nautilus: -> dark
5. hypr keybind -> wofi -> foot-client [hypr env not set] -> nautilus: -> light
6. hypr keybind -> wofi -> foot-client [hypr env not set] -> export GTK_THEME=Adwaita-dark -> nautilus: -> dark
```

looks like programs started by hyprland do have the environment set, but `waybar`, `wofi` not seems to export the env to the applications it runs? (but foot launched from wofi do have the env set?)

**Reply by Aqa-Ib** [reply]
why do you use `GTK_THEME` instead of calling gsettings ?

**Reply by rtgiskard** [reply]
> why do you use `GTK_THEME` instead of calling gsettings ?

I do have set the dark theme with gsettings, though it's not write as `exec-once`.
```
$ gsettings list-recursively org.gnome.desktop.interface | grep -i dark
org.gnome.desktop.interface color-scheme 'prefer-dark'
org.gnome.desktop.interface gtk-theme 'Adwaita-dark'
```

**Reply by Aqa-Ib** [reply]
Why do you need the env GTK_THEME then? I mean, you are setting it globally with gsettings. Are you using the env for a per-application theming?

**Reply by rtgiskard** [reply]
> Why do you need the env GTK_THEME then? I mean, you are setting it globally with gsettings. Are you using the env for a per-application theming?

Dark theme once set by `gsettings` not seems to work effective, then I have to set `GTK_THEME`

**Reply by Aqa-Ib** [reply]
You can call gsettings as many times as you want. You can use it to change from dark to light themes instantly. I recommend you doing that instead of using the environment variable.

**Reply by rtgiskard** [reply]
> You can call gsettings as many times as you want. You can use it to change from dark to light themes instantly. I recommend you doing that instead of using the environment variable.

But without the extra `GTK_THEME`, some app like nautilus may run with the light theme, no matter how many times I call `gsettings` manually, which should only touch the gtk config files 😂

**Reply by rtgiskard** [reply]
And just add the two lines "exec = gsettings .." from above and restart hyprland, if `GTK_THEME` not set, nautilus will allways run with light theme

**Reply by rtgiskard** [reply]
Well, looks like it's related to `xdg-desktop-portal-gtk`, the `gsettings` solution works well with gtk portal with no need to set any environment variable.

I once removed `xdg-desktop-portal-gtk` and only have `xdg-desktop-portal-hyprland` in the system

**Comment by sallymanda** [comment]
Thanks @Aqa-Ib for the suggestion to use `gsettings` for GTK theme changing; I got it automated for day and night switching and am loving it. But wondering how to achieve the same for my qt apps, anyone know how to change qt themes via command line? I've been using a qt6ct fork to change breeze themes manually, but it doesnt support cli arguments...

**Reply by Aqa-Ib** [reply]
You have to use `sed` to change the theme in the qt6ct config file. Later I will send you the command that I use for this.

**Reply by Aqa-Ib** [reply]
Qt
+ Command for color change:
`  sed -i "s/color_scheme_path=.*/color_scheme_path=\"$qtColors\"/" ~/.config/qt6ct/qt6ct.conf`
+ Command for icon change:
`  sed -i "s/icon_theme=.*/icon_theme=$iconTheme/" ~/.config/qt6ct/qt6ct.conf`

Substitute $qtColors and $iconTheme with your chosen one. The change is instant, meaning that when a Qt app is open, it will change colors without needing to restart it.

Have fun !

**Reply by sallymanda** [reply]
Thanks! Hells yeah, finally my Hyprland theming feels as smooth as a desktop environment! For anyone who is a newbie like me, you have to replace `$qtColors` with the full path with escaped slashes, so for me just using the Breeze colors I did the following:

`sed -i "s/color_scheme_path=.*/color_scheme_path=\"\/usr\/share\/color-schemes\/BreezeLight.colors\"/" ~/.config/qt6ct/qt6ct.conf`

And this for dark:

`sed -i "s/color_scheme_path=.*/color_scheme_path=\"\/usr\/share\/color-schemes\/BreezeDark.colors\"/" ~/.config/qt6ct/qt6ct.conf`

**Reply by major-mayer** [reply]
Thanks for the scripts. 
In case anybody want's to apply this on Manjaro/ Arch-based systems with Gnome desktop environment, here are the commands adjusted for Arch's file structure (extra quotation marks around the path don't seem to be required for .conf files):

Color scheme Darker (Night mode)
```
sed -i "s/color_scheme_path=.*/color_scheme_path=\/usr\/share\/qt6ct\/colors\/darker.conf/" ~/.config/qt6ct/qt6ct.conf
```

Color scheme Simple (Light mode)
```
sed -i "s/color_scheme_path=.*/color_scheme_path=\/usr\/share\/qt6ct\/colors\/simple.conf/" ~/.config/qt6ct/qt6ct.conf
```

BTW: I use this Gnome extension to execute the scripts: https://nightthemeswitcher.romainvigier.fr/

**Reply by major-mayer** [reply]
I just noticed that these color scheme configuration files are actually slighty different colors than the Breeze ones.
However, when I try to use the Breeze `.color` files, qt6ct doesn't seem to apply them correctly (or rather at all).
Because the structure of these files is also pretty different, I think they are not compatible with qt6ct (anymore?).
How did you guys get this to work?

**Reply by sallymanda** [reply]
I'm using [this fork](https://aur.archlinux.org/packages/qt6ct-kde) if it helps ;-)

**Reply by major-mayer** [reply]
Great hint, thank you very much. With this fork, applying the Breeze color scheme files work perfectly well.

**Comment by PILIHU2022** [comment]
Thanks you all! Have nice day! 
I am sorry to reply you too late

**Comment by nzlov** [comment]
Install `yin-yang` by aur.

## `hyprland.conf`
```
exec-one=yin-yang

env = QT_QPA_PLATFORMTHEME,qt6ct
```

## yin-yang
Plugins enbable Custom
```
gsettings set org.gnome.desktop.interface color-scheme "prefer-light" && sed -i "s/darker/airy/" ~/.config/qt6ct/qt6ct.conf
```
and
```
gsettings set org.gnome.desktop.interface color-scheme "prefer-dark" && sed -i "s/airy/darker/" ~/.config/qt6ct/qt6ct.conf
```

**Reply by PILIHU2022** [reply]
This problem has been solved. Anyway, Thank you for your answer!
:)

**Reply by sudg6** [reply]
how?

**Comment by Fuzzypyro** [comment]
For anyone who ended up here for the same problem I fixed it by adding this dependency.

xdg-desktop-portal-wlr

**Comment by prashantrahul141** [comment]
For anyone wondering how to achieve the same using home-manager, you can use [dconf options:](https://nix-community.github.io/home-manager/options.xhtml#opt-dconf.settings)
```
dconf.settings = {
   "org/gnome/desktop/interface" = {
     color-scheme = "prefer-dark";
   };
};
```

**Comment by PILIHU2022** [comment]
https://github.com/PILIHU2022/My-dotfiles/blob/main/How-to-set-dark-mode.md You can read this document written in Chinese. It's on **Arch Linux**. If you can't understand Chinese, please translate it by Google. I am sorry that I can't reply the answer because I should study for a very important test. Hope it works!

________________________________

**Comment by Septias** [comment]
On Nixos 25.05, I noticed that dark mode setting broke for me. To fix it, it basically came down to fixing the [portals](https://flatpak.github.io/xdg-desktop-portal/) and ensuring dconf is running (gsettings is not needed as it is only a frontend to dconf). For NixOS, it is also important to make the distinction between home-manager and config-based installation. I installed through home-manager, and then it was not possible to fix the portals in `configuration.nix` because the home-manager-defined portals took precedence. 

I now have this in my home-manager configuration:
```
  xdg.enable = true;
  xdg.portal = {
    enable = true;
    extraPortals = [pkgs.xdg-desktop-portal-gtk pkgs.xdg-desktop-portal-hyprland pkgs.xdg-desktop-portal-wlr pkgs.xdg-desktop-portal-gnome];
    configPackages = [pkgs.xdg-desktop-portal-gtk pkgs.xdg-desktop-portal-hyprland pkgs.xdg-desktop-portal-wlr pkgs.xdg-desktop-portal-gnome];
    config.common = {
      default = ["gnome" "hyprland" "gtk"];
      "org.freedesktop.impl.portal.Settings" = "gnome";
    };
  };
```
and it seems to work. This seems to be a reasonable check to see if things are working:

```
dconf read /org/gnome/desktop/interface/color-scheme
```

**Comment by sephid86** [comment]
아치리눅스에서 dconf write /org/gnome/desktop/interface/color-scheme 'prefer-dark' 하고 gnome-themes-extra 설치하니까 해결 되었습니다.

I managed to fix the issue on Arch Linux by running the command dconf write /org/gnome/desktop/interface/color-scheme 'prefer-dark' and installing the gnome-themes-extra package.

**Comment by nghiatr84** [comment]
https://wiki.archlinux.org/title/Dark_mode_switching
https://wiki.archlinux.org/title/Qt#Configuration_of_Qt_5/6_applications_under_environments_other_than_KDE_Plasma
These pages helped me a lot.
